pkg/migration-tool/reader/reader.go

```diff
@@ -0,0 +1,106 @@
+package reader
+
+import (
+	"context"
+	"fmt"
+	"go.uber.org/atomic"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/timescale/promscale/pkg/log"
+	plan "github.com/timescale/promscale/pkg/migration-tool/planner"
+	"github.com/timescale/promscale/pkg/migration-tool/utils"
+)
+
+const defaultReadTimeout = time.Minute * 5
+
+type RemoteRead struct {
+	c               context.Context
+	url             string
+	client          *utils.Client
+	plan            *plan.Plan
+	fetchersRunning atomic.Uint32
+	sigBlockRead    chan *plan.Block // To the writer.
+	SigForceStop    chan struct{}
```

Comment:
I can't find the code that feeds this channel

Reply:
This is used inside the integration tests so as to introduce halts while migration. For introducing any halt, just stop the reader that that's where this channel finds its use-case.